src/LogViewer/index.js

```diff
@@ -106,6 +114,17 @@ export default class LogViewer extends React.Component {
   handleMessage(e) {
     if (e.data && typeof e.data === 'object') {
       this.setState({ data: e.data });
+
+      Object.keys(e.data).forEach((name) => {
+        switch(name) {
+          case 'highlightError':
+            return this.highlightError(e.data.highlightError);
+          case 'lineNumber':
+            return this.jumpToLine(e.data.lineNumber);
+          case 'highlightStart':
+            return this.setState({ highlightStart: e.data.highlightStart, highlightEnd: e.data.highlightEnd });
```

Comment:
I could replicate this, but I made a few changes and it seems to work OK now. I'll put some notes in the review comment.